orttraining/orttraining/python/ort_trainer.py

```diff
@@ -691,6 +693,73 @@ def load_state_dict(self, state_dict, strict=False):
             session_state[name] = state_dict[name].numpy()
         self.session.load_state(session_state, strict)
 
+    def save_checkpoint(self, checkpoint_dir, checkpoint_prefix="ORT_checkpoint", checkpoint_state_dict=None):
+        if checkpoint_state_dict==None:
+            checkpoint_state_dict={'model': self.state_dict()}
+        else:
+            checkpoint_state_dict.update({'model': self.state_dict()})
+
+        if not os.path.exists(checkpoint_dir):
+            os.makedirs(checkpoint_dir)
+
+        if self.partition_optimizer_:
+            checkpoint_name = checkpoint_prefix + "_ZeROrank_" + str(self.world_rank) + "_of_" + str(self.world_size-1) + ".tar"
+        else:
+            checkpoint_name = checkpoint_prefix + ".tar" 
+        checkpoint_file = os.path.join(checkpoint_dir, checkpoint_name)
+
+        if os.path.exists(checkpoint_file):
+            print("WARNING: {} already exists, overwriting.".format(checkpoint_file))
+
+        print("Saving checkpoint {}".format(checkpoint_file))
+        torch.save(checkpoint_state_dict, checkpoint_file)
```

Comment:
The motivation for '.tar' was that pytorch recommends a checkpoint having additional info other than the model weights to be saved as '.tar' 
Source: https://pytorch.org/tutorials/beginner/saving_loading_models.html
I agree that it can be confusing, and we can save as '.ort.pt'